src/tslint/noSingleDeclareModuleRule.ts

```diff
@@ -0,0 +1,53 @@
+import * as Lint from "tslint/lib/lint";
+import * as ts from "typescript";
+
+export class Rule extends Lint.Rules.AbstractRule {
+	static metadata: Lint.IRuleMetadata = {
+		ruleName: "no-single-declare-module",
+		description: "Don't use an ambient module declaration if you can use an external module file.",
+		rationale: "Cuts down on nesting",
+		options: {},
+		type: "style"
+	};
+
+	static FAILURE_STRING = "File has only 1 module declaration — write it as an external module.\n" +
+		"If augmenting a single module, use `// tslint:disable-next-line:no-single-declare-module`.";
```

Comment:
I do not think this is needed. just make sure the file is not a module before you call applyWithWalker.